app/src/main/java/com/exallium/mvvmcleanexample/app/users/UserEditView.kt

```diff
@@ -25,34 +26,46 @@ class UserEditView : AppCompatActivity() {
     @Inject
     lateinit var router: AppRouter
 
-    val compositeDisposable = CompositeDisposable()
+    private var binding by autoEcoBinding<UserEditViewBinding>(null)
+    private var disposable by ecoDisposable(null)
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        val binding = DataBindingUtil.setContentView<UserEditViewBinding>(this, R.layout.user_edit_view)
+        binding = DataBindingUtil.setContentView<UserEditViewBinding>(this, R.layout.user_edit_view).also { binding ->
+            DaggerUserEditComponent.builder()
+                    .appComponent(App.component)
+                    .domainUserModule(DomainUserModule())
+                    .userEditModule(UserEditModule())
+                    .build()
+                    .inject(this)
 
-        DaggerUserEditComponent.builder()
-                .appComponent(App.component)
-                .domainUserModule(DomainUserModule())
-                .userEditModule(UserEditModule())
-                .build()
-                .inject(this)
+            binding.viewModel = viewModel
 
-        binding.viewModel = viewModel
-
-        router.transitions().subscribe {
-            when (it) {
-                is AppRouter.Transition.GoBack -> onBackPressed()
-                is AppRouter.Transition.DisplayMessage -> displayMessage(it.message)
-                is AppRouter.Transition.DisplayError -> displayError(binding.root, it.message)
+            router.transitions().subscribe {
+                when (it) {
+                    is AppRouter.Transition.GoBack -> {/*FIXME this causes a crash: onBackPressed()*/}
```

Comment:
It's supposed to "go back" to the previous screen.  in this case, that'd be onBackPressed.